extensions/guacamole-auth-radius/src/main/java/org/apache/guacamole/auth/radius/RadiusConnectionService.java

```diff
@@ -0,0 +1,394 @@
+                                            confService.getRadiusSharedSecret(),
+                                            confService.getRadiusAuthPort(),
+                                            confService.getRadiusAcctPort(),
+                                            confService.getRadiusTimeout());
+        }
+        catch (GuacamoleException e) {
+            logger.error("Unable to initialize RADIUS client: {}", e.getMessage());
+            logger.debug("Failed to init RADIUS client.", e);
+            return;
+        }
+        catch (UnknownHostException e) {
+            logger.error("Unable to resolve host: {}", e.getMessage());
+            logger.debug("Failed to resolve host.", e);
+            return;
+        }
+        catch (IOException e) {
+            logger.error("Unable to communicate with host: {}", e.getMessage());
+            logger.debug("Failed to communicate with host.", e);
+            return;
+        }
+
+    }
+
+    /**
+     * Creates a new instance of RadiusAuthentictor, configured with
+     * parameters specified within guacamole.properties.
+     *
+     * @return
+     *     A new RadiusAuthenticator instance which has been configured
+     *     with parameters from guacamole.properties, or null if
+     *     configuration fails.
+     *
+     */
+    private RadiusAuthenticator setupRadiusAuthenticator() {
+
+        // If we don't have a radiusClient object, yet, don't go any further.
+        if (radiusClient == null) {
+            logger.error("RADIUS client hasn't been set up, yet.");
+            logger.debug("We can't run this method until the RADIUS client has been set up.");
+            return null;
+        }
+
+        String radAuthName;
+        String caFile;
+        String caPassword;
+        String caType;
+        String keyFile;
+        String keyPassword;
+        String keyType;
+        Boolean trustAll;
+        String innerProtocol;
+        LocalEnvironment guacEnv;
+        String basePath;
+
+        // Pull configuration parameters from guacamole.properties
+        try {
+            guacEnv = new LocalEnvironment();
+            basePath = guacEnv.getGuacamoleHome().getAbsolutePath() + '/';
+            radAuthName = confService.getRadiusAuthProtocol();
+            caFile = confService.getRadiusCAFile();
+            caPassword = confService.getRadiusCAPassword();
+            caType = confService.getRadiusCAType();
+            keyFile = confService.getRadiusKeyFile();
+            keyPassword = confService.getRadiusKeyPassword();
+            keyType = confService.getRadiusKeyType();
+            trustAll = confService.getRadiusTrustAll();
+            innerProtocol = confService.getRadiusEAPTTLSInnerProtocol();
+            
+        }
+        catch (GuacamoleException e) {
+            logger.error("Error retrieving configuration.");
+            logger.debug("Error getting config parameters from file.");
+            return null;
```

Comment:
Why not allow this function to throw `GuacamoleException` rather than masking it?